gpytorch/lazy/lazy_tensor.py

```diff
@@ -1422,57 +1643,73 @@ def root_inv_decomposition(self, initial_vectors=None, test_vectors=None):
-                raise RuntimeError(
-                    "LazyTensor (size={}) cannot be multiplied with initial_vectors (size={}).".format(
-                        self.shape, initial_vectors.shape
-                    )
-                )
 
-        inv_roots = self._root_inv_decomposition(initial_vectors)
+            inv_roots = self._root_inv_decomposition(initial_vectors)
 
-        # Choose the best of the inv_roots, if there were more than one initial vectors
-        if initial_vectors is not None and initial_vectors.size(-1) > 1:
-            num_probes = initial_vectors.size(-1)
-            test_vectors = test_vectors.unsqueeze(0)
+            # Choose the best of the inv_roots, if there were more than one initial vectors
+            if initial_vectors is not None and initial_vectors.size(-1) > 1:
+                num_probes = initial_vectors.size(-1)
+                test_vectors = test_vectors.unsqueeze(0)
 
-            # Compute solves
-            solves = inv_roots.matmul(inv_roots.transpose(-1, -2).matmul(test_vectors))
+                # Compute solves
+                solves = inv_roots.matmul(inv_roots.transpose(-1, -2).matmul(test_vectors))
 
-            # Compute self * solves
-            solves = (
-                solves.permute(*range(1, self.dim() + 1), 0)
-                .contiguous()
-                .view(*self.batch_shape, self.matrix_shape[-1], -1)
-            )
-            mat_times_solves = self.matmul(solves)
-            mat_times_solves = mat_times_solves.view(*self.batch_shape, self.matrix_shape[-1], -1, num_probes).permute(
-                -1, *range(0, self.dim())
-            )
+                # Compute self * solves
+                solves = (
+                    solves.permute(*range(1, self.dim() + 1), 0)
+                    .contiguous()
+                    .view(*self.batch_shape, self.matrix_shape[-1], -1)
+                )
+                mat_times_solves = self.matmul(solves)
+                mat_times_solves = mat_times_solves.view(
+                    *self.batch_shape, self.matrix_shape[-1], -1, num_probes
+                ).permute(-1, *range(0, self.dim()))
 
-            # Compute residuals
-            residuals = (mat_times_solves - test_vectors).norm(2, dim=-2)
-            residuals = residuals.view(residuals.size(0), -1).sum(-1)
+                # Compute residuals
+                residuals = (mat_times_solves - test_vectors).norm(2, dim=-2)
+                residuals = residuals.view(residuals.size(0), -1).sum(-1)
 
-            # Choose solve that best fits
-            _, best_solve_index = residuals.min(0)
-            inv_root = inv_roots[best_solve_index].squeeze(0)
+                # Choose solve that best fits
+                _, best_solve_index = residuals.min(0)
+                inv_root = inv_roots[best_solve_index].squeeze(0)
 
-        else:
-            inv_root = inv_roots
+            else:
+                inv_root = inv_roots
```

Comment:
this is a pretty big if block. For better readability / avoiding huge indents, how about moving this into a separate function and just call that here?